QSV: Erroneous log message

@galinart 

When running an encode on a non-intel system, we get an erroneous log message:

qsv: hb_qsv_get_platform incorrect qsv device index 0

Seems to be generated just after starting an encode in the "activity_log_main" in the main ui.

HandBrake Nightly 20210321222136-126013c87-master (2021032201)
OS: Microsoft Windows NT 10.0.19042.0
CPU: AMD Ryzen 9 5900X 12-Core Processor            
Ram: 32711 MB, 
GPU Information:
  NVIDIA GeForce GTX 1060 6GB - 27.21.14.6172
Screen: 2560x1440
Temp Dir: C:\Users\Scott\AppData\Local\Temp\
Install Dir: H:\Development\HandBrake\Dev\win\CS\HandBrakeWPF\bin\Debug\
Data Dir: C:\Users\Scott\AppData\Roaming\HandBrake\Nightly

-------------------------------------------

[22:22:56] hb_init: starting libhb thread

 # Starting Scan ...

[22:22:56] CPU:
[22:22:56]  - logical processor count: 24
[22:22:56] Intel Quick Sync Video support: no
[22:22:56] hb_scan: path=H:\Development\HandBrake\Samples\bbb_sunflower_1080p_60fps_normal.mp4, title_index=0
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:323: failed opening UDF image H:\Development\HandBrake\Samples\bbb_sunflower_1080p_60fps_normal.mp4
src/libbluray/disc/disc.c:424: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:424: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2585: nav_get_title_list(H:\Development\HandBrake\Samples\bbb_sunflower_1080p_60fps_normal.mp4\) failed
[22:22:56] bd: not a bd - trying as a stream/file instead
libdvdnav: Using dvdnav version 6.0.1
libdvdread: Encrypted DVD support unavailable.
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed
libdvdread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: failed to read VIDEO_TS.IFO
[22:22:56] dvd: not a dvd - trying as a stream/file instead
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'H:\Development\HandBrake\Samples\bbb_sunflower_1080p_60fps_normal.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 1
    compatible_brands: isomavc1
    creation_time   : 2013-12-16T17:59:32.000000Z
    title           : Big Buck Bunny, Sunflower version
    artist          : Blender Foundation 2008, Janus Bager Kristensen 2013
    comment         : Creative Commons Attribution 3.0 - http://bbb3d.renderfarming.net
    genre           : Animation
    composer        : Sacha Goedegebure
  Duration: 00:10:34.53, start: 0.000000, bitrate: 4486 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], 4001 kb/s, 60 fps, 60 tbr, 60k tbn, 120 tbc (default)
    Metadata:
      creation_time   : 2013-12-16T17:59:32.000000Z
      handler_name    : GPAC ISO Video Handler
    Stream #0:1(und): Audio: mp3 (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 160 kb/s (default)
    Metadata:
      creation_time   : 2013-12-16T17:59:37.000000Z
      handler_name    : GPAC ISO Audio Handler
    Stream #0:2(und): Audio: ac3 (ac-3 / 0x332D6361), 48000 Hz, 5.1(side), fltp, 320 kb/s (default)
    Metadata:
      creation_time   : 2013-12-16T17:59:37.000000Z
      handler_name    : GPAC ISO Audio Handler
    Side data:
      audio service type: main
[22:22:56] scan: decoding previews for title 1
[22:22:57] scan: audio 0x1: mp3float, rate=48000Hz, bitrate=160000 Unknown (MP3) (2.0 ch) (160 kbps)
[22:22:57] scan: audio 0x2: ac3, rate=48000Hz, bitrate=320000 Unknown (AC3) (5.1 ch) (320 kbps)
[22:22:57] scan: 10 previews, 1920x1080, 60.000 fps, autocrop = 0/0/0/0, aspect 16:9, PAR 1:1, color profile: 1-1-1
[22:22:57] libhb: scan thread found 1 valid title(s)

 # Processing Scan Information ...


 # Scan Finished ...

**_qsv: hb_qsv_get_platform incorrect qsv device index 0_**

--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -2857,7 +2857,6 @@ int hb_qsv_get_platform(int adapter_index)
             return qsv_map_mfx_platform_codename(info->Platform.CodeName);
         }
     }
-    hb_error("qsv: hb_qsv_get_platform incorrect qsv device index %d", adapter_index);
     return HB_CPU_PLATFORM_UNSPECIFIED;
 }
 

--- a/win/CS/HandBrakeWPF/Services/Encode/Factories/EncodeTaskFactory.cs
+++ b/win/CS/HandBrakeWPF/Services/Encode/Factories/EncodeTaskFactory.cs
@@ -255,7 +255,7 @@ namespace HandBrakeWPF.Services.Encode.Factories
             
             video.Options = job.ExtraAdvancedArguments;
 
-            if (HandBrakeHardwareEncoderHelper.QsvHardwareGeneration > 6 && (job.VideoEncoder == VideoEncoder.QuickSync || job.VideoEncoder == VideoEncoder.QuickSyncH265 || job.VideoEncoder == VideoEncoder.QuickSyncH26510b))
+            if (HandBrakeHardwareEncoderHelper.IsQsvAvailable && (HandBrakeHardwareEncoderHelper.QsvHardwareGeneration > 6) && (job.VideoEncoder == VideoEncoder.QuickSync || job.VideoEncoder == VideoEncoder.QuickSyncH265 || job.VideoEncoder == VideoEncoder.QuickSyncH26510b))
             {
                 if (configuration.EnableQsvLowPower && !video.Options.Contains("lowpower"))
                 {

